### DISCUSSION TITLE
How to: Fully supervised learning on a single graph?

### DISCUSSION THREAD

**Discussion by a-0-ls** [pyg-team/pytorch_geometric#4940] [Q&A]
Hi,

A very general question. If I have a single graph, is it possible to do supervised learning instead of using masks to do semi- supervised learning? 

In this case, how can I make my test dataset?
I have networkx data object as my input graph.  How can I get the test set out of this?

I have seen the randomlinksplit and randomnodesplit, my use case is for solving a node regression problem.
Any feedback would be very helpful.

**Comment by wsad1** [comment]
First you'll have to convert your graph to a `torch_geometric.data` instance using [from_networkx](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.from_networkx).

Then, you could use [subgraph](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.subgraph) to split your graph into separate train and test graphs (subgraphs).
